### PR TITLE
Fix parallel builds and builds against old GLib

### PR DESCRIPTION
**Pull request by sunweaver**

### CODE REVIEW

File: src/frontend/meson.build [hunk 1]
```diff
@@ -30,6 +30,7 @@ libfrontend_dependencies = [
 
 libfrontend_includes = [
     include_directories('.'),
+    include_directories('../../data'),
```
Comment (ikeydoherty) [inline src/frontend/meson.build]: This seems.. wrong..

**Comment by sunweaver** [comment]
On  Fr 19 Jan 2018 23:22:43 CET, Ikey Doherty wrote:

> ikeydoherty commented on this pull request.
>
>
>
>> @@ -30,6 +30,7 @@ libfrontend_dependencies = [
>
>  libfrontend_includes = [
>      include_directories('.'),
> +    include_directories('../../data'),
>
> This seems.. wrong..

It makes the build find the .c and .h files related that get created  
in data/ by GDBus codegen.


-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by ikeydoherty** [comment]
Oh I know what it does but it just seems a bit .. wrong.. including a data directory into source.

**Comment by sunweaver** [comment]
On  Fr 19 Jan 2018 23:37:21 CET, Ikey Doherty wrote:

> Oh I know what it does but it just seems a bit .. wrong.. including  
> a data directory into source.

I fully get that. The concept of the PR is to move the codegen part to  
the very beginning of the build, before the parallel builds of the  
subdirs steps in. As a non-meson person (for now), this was the  
approach that made things work here.

-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by ikeydoherty** [comment]
What version of meson/glib are you on? i.e checking where the bug actually lies here because on newer distros there is no build issue (i.e. Solus, Arch, etc)

**Comment by ikeydoherty** [comment]
FWIW I think I'm gonna have to merge it anyway (ty :)) but I'd still like to understand it better. I dislike black boxes

**Comment by sunweaver** [comment]
On  Fr 19 Jan 2018 23:46:13 CET, Ikey Doherty wrote:

> What version of meson/glib are you on? i.e checking where the bug  
> actually lies here because on newer distros there is no build issue  
> (i.e. Solus, Arch, etc)

I am on Debian unstable (and see an asset in backporting to stretch).  
Check these URLs for looking at the currently available versions there  
in unstable/sid and stretch-backports:

https://packages.debian.org/source/sid/meson
https://packages.debian.org/source/sid/glib2.0

Please note that the parallel builds issue is a race condition thing.  
It does not occur often.

See the old logs (click the links) here:  
https://buildd.debian.org/status/package.php?p=brisk-menu

Here are the patches I added to 0.5.0:
https://anonscm.debian.org/cgit/pkg-mate/brisk-menu.git/tree/debian/patches?id=bd08d66f09ad39798c11c9a7fe9c103487b5344c

Those patches should be the same I filed as this PR, you may want to  
check that they are identical, because those are the tested-a-lot  
patches.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by ikeydoherty** [comment]
Still builds here, so thats just how its gonna need to be. Thanks for the changes bud!

**Pull request merged**